### PR TITLE
chore: Project GraphQL

### PR DESCRIPTION
**Pull request by cbolles**
* Change out Project module for GraphQL usage

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
     "test:watch": "ng test",
     "lint:fix": "ng lint --fix",
     "lint": "ng lint",
-    "setup": "node setup.js"
+    "setup": "node setup.js",
+    "generate-graphql": "graphql-codegen --config graphql-codegen.yml"
```
Comment (gregfrasco) [inline client/package.json:13]: Nit: This is process is called introspection. Might want to call it that as it the same in other projects.

File: client/src/app/core/services/project.service.ts [hunk 2]
```diff
@@ -1,20 +1,40 @@
...
-  constructor(private readonly signLab: SignLabHttpClient) {
-    this.updateProjects();
+  constructor(
+    projectsGQL: GetProjectsGQL,
+    private readonly createProjectGQL: CreateProjectGQL
+  ) {
+    // Subscribe to the project query
+    this.projectQuery = projectsGQL.watch();
+    this.projectQuery.valueChanges.subscribe((result) => {
+      this.projectsObs.next(result.data.getProjects);
```
Comment (gregfrasco) [inline client/src/app/core/services/project.service.ts:36]: oh observables

File: server/src/generate-schema.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { NestFactory } from '@nestjs/core';
```
Comment (gregfrasco) [inline server/src/generate-schema.ts:1]: You have an auto scheme file do you need a separate entry point to create the scheme? Unless you are no committing the scheme?

File: server/src/project/project.module.ts [hunk 4]
```diff
@@ -1,19 +1,21 @@
...
+import { ProjectChangePipe } from './project.dto';
 
 @Module({
   imports: [
     MongooseModule.forFeature([{ name: Project.name, schema: ProjectSchema }]),
-    AuthModule,
-    UserModule,
+    forwardRef(() => AuthModule),
+    forwardRef(() => UserModule),
+    forwardRef(() => SharedModule),
```
Comment (gregfrasco) [inline server/src/project/project.module.ts:16]: fun

**Review by gregfrasco — APPROVED** [review]

File: server/src/project/project.module.ts (hunk 4 above)
Comment (cbolles) [reply server/src/project/project.module.ts:16]: Yea this physically pains me

**Pull request merged**